source/src/EventSource.cc

```diff
@@ -0,0 +1,287 @@
+      
+      CollectorInfo info;
+      info.m_registered = false;
+      m_collectorInfos.insert(CollectorInfoMap::value_type(name, info));
+    }
+
+    //-------------------------------------------------------------------------------------------------
+    
+    void EventSource::start()
+    {
+      if(m_started)
+        throw core::StatusCodeException(core::STATUS_CODE_ALREADY_INITIALIZED);
+      
+      auto streamerPtr = PluginManager::instance()->create<core::EventStreamer>(m_streamerName);
+      m_eventStreamer = std::shared_ptr<core::EventStreamer>(streamerPtr);
+      
+      if(!m_eventStreamer)
+      {
+        dqm_error( "EventSource::start(): Event streamer '{0}' not found in available streamers. Couldn't initialize!", m_streamerName );
+        throw core::StatusCodeException(core::STATUS_CODE_NOT_FOUND);        
+      }
+      
+      if(m_collectorInfos.empty())
+      {
+        dqm_error( "EventSource::start(): Event source '{0}' has no collector to send event. Couldn't initialize!", m_sourceName );
+        throw core::StatusCodeException(core::STATUS_CODE_NOT_INITIALIZED);
+      }
+      
+      m_bufferDevice = std::make_shared<xdrstream::BufferDevice>(2*1024*1024);
+      
+      Json::Value sourceInfo;
+      this->getSourceInfo(sourceInfo);
+      
+      for(auto colIter : m_collectorInfos)
+      {
+        bool registered = this->registerMe(colIter.first, sourceInfo);
+        colIter.second.m_registered = registered;
+      }
+      
+      m_started = true;
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    void EventSource::sendEvent(const Event *const pEvent)
+    {
+      core::StringVector collectors;
+
+      for(auto iter : m_collectorInfos)
+        collectors.push_back(iter.first);
+        
+      this->sendEvent(collectors, pEvent);
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    void EventSource::sendEvent(const std::string &collector, const Event *const pEvent)
+    {
+      if(m_collectorInfos.end() == m_collectorInfos.find(collector))
+      {
+        dqm_error( "EventSource::sendEvent(col,evt): collector '{0}' not registered !", collector );
+        throw core::StatusCodeException(core::STATUS_CODE_NOT_FOUND);
+      }
+      
+      this->sendEvent(core::StringVector({collector}), pEvent);
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    void EventSource::sendEvent(const core::StringVector &collectors, const Event *const pEvent)
+    {
+      if(!m_started)
+        throw core::StatusCodeException(core::STATUS_CODE_NOT_INITIALIZED);
+        
+      if(nullptr == pEvent)
+        throw core::StatusCodeException(core::STATUS_CODE_INVALID_PTR);
+      
+      (void)m_bufferDevice->reset();
```

Comment:
Genuine question: Why is the `(void)` needed here?

Reply:
To make some compiler happy as it normally returns a status code